terra/contracts/token-bridge/src/contract.rs

```diff
@@ -247,6 +283,27 @@ pub fn execute(deps: DepsMut, env: Env, info: MessageInfo, msg: ExecuteMsg) -> S
             recipient_chain,
             recipient.as_slice().to_vec(),
             fee,
+            TransferType::WithoutPayload,
+            nonce,
+        ),
+        ExecuteMsg::InitiateTransferWithPayload {
+            asset,
+            recipient_chain,
+            recipient,
+            fee,
+            payload,
+            nonce,
+        } => handle_initiate_transfer(
+            deps,
+            env,
+            info,
+            asset,
+            recipient_chain,
+            recipient.as_slice().to_vec(),
+            fee,
+            TransferType::WithPayload {
+                payload: payload.as_slice().to_vec(),
```

Comment:
This is making an unnecessary copy.  We can just do `payload: payload.into()`, which will unwrap the `Binary` and return the inner `Vec<u8>`.

Reply:
When you have time, check out the latest push to this branch. I changed this and others to use `into()`.